ql/src/test/queries/clientpositive/cbo_cte_materialization.q

```diff
@@ -0,0 +1,28 @@
+--! qt:dataset:src
+
+set hive.optimize.cte.materialize.threshold=1;
+set hive.optimize.cte.materialize.full.aggregate.only=false;
```

Comment:
```
set hive.query.planmapper.link.relnodes=false;
```
By setting this property the test fails with the same `RuntimeException: equivalence mapping violation`.

Reply:
I'm checking the whole execution and what the flag does. Disabling `hive.query.planmapper.link.relnodes`, no aux signature is created, and no merge across Operators doesn't happen. I feel it is not consistent with the description, `Whether to link Calcite nodes to runtime statistics`.
I guess the direct problem will be resolved if we skip linking RelNodes with Operators when `hive.query.planmapper.link.relnodes=false` is configured. That sounds more consistent with the description.

I'm putting my additional notes here. I tried to put some to-be solutions but it could not be very easy.
https://docs.google.com/document/d/1LCST23cSBZglBzjhnCqHlpcLv6xrXRBxU_wszSDAk9w/edit?usp=sharing

Reply:
I guess `hive.query.planmapper.link.relnodes` can be implemented like this. @kgyrtkirk might have some more contexts.
https://github.com/okumin/hive/commit/cdab2c1941411caa9cef45c2821689982cb2dfe3

Reply:
I found this fails if we disable `hive.query.planmapper.link.relnodes`.

```
create table src (key string, value string);

set hive.cbo.fallback.strategy=NEVER;
set hive.query.planmapper.link.relnodes=false;
set hive.optimize.ppd=false;

EXPLAIN
SELECT * FROM src WHERE key = '5'
UNION ALL
SELECT * FROM src WHERE key = '5';
```

@zabetak As a potential workaround, I'm wondering if it makes sense to relax the validation and disable features with which PlanMapper is involved when "equivalence mapping violation" happens.
I presume PlanMapper succeeds in 99.9% cases as most qtests succeed. Currently, the remaining 0.1% fails and it sounds like an excessive penalty. We can keep the validation for qtests so that we can minimize the risk of degradation.
https://github.com/okumin/hive/commit/dd5be1543b9426f5d6ab94a37d5cdf8136c5fec7